src/evaluate.py

```diff
@@ -0,0 +1,72 @@
+import pandas as pd
+import numpy as np
+import re
+from src.model import yoModel
+from tqdm import tqdm
+from typing import List, Tuple
+from src.utils import WORDS_REGEX
+from dataclasses import dataclass
+from sklearn.metrics import accuracy_score, precision_score, recall_score, \
+    f1_score, roc_auc_score
+
+
+@dataclass
+class EvaluateResult:
+    """Model evaluation result."""
+    accuracy: float
+    precision: float
+    recall: float
+    f1_score: float
+    auroc: float
+
+    def __str__(self):
+        return "\n".join(["{}: {}".format(m, v) for m, v in self.__dict__.items()])
```

Comment:
Вот здесь необязательно использовать приватное поле `__dict__`. 
В `dataclasses` есть функция [`asdict()`](https://docs.python.org/3.8/library/dataclasses.html#dataclasses.asdict), которая скорее всего будет по умолчанию гораздо лучше покрывать конвертацию в словарь.

Кроме того, предлагаю использовать f-строки! Все же для Python 3.8+ (который в проекте предполагается), они считаются де-факто стандартом и на практике экономят место в смысле написанного кода.

```suggestion
        return "\n".join([f"{m}: {v}" for m, v in asdict(self).items()])
```